### ISSUE TITLE
Advanced options selections not being passed on to R server

### ISSUE THREAD

**Issue by benthestatistician** [RTICWDT/evaluation-engine-website#4]
I tried to request within-district matching but didn't wind up getting it.  I think the issue is on the front-end, with the special instructions not being passed along, although I couldn't confirm this due to an unrelated problem.  

Perhaps relatedly, when I select a non-default district matching option, hit Save & Continue from within the Advanced Options screen, then hit Advanced Options again once I'm routed back to the main options page, the radio button again shows the default choice for district matching, not the option I had previously selected.

**Comment by benthestatistician** [comment]
I was able to confirm that in the job in question (2c73b5d9-e94b-48e0-aeda-22dd1a9d6d0d, on development), the within-district matching preference that I thought I had registered in the Advanced Options screen did not make it into the job spec.

**Comment by rstudley** [comment]
@markmfredrickson Passing this along from Alex: As for the advanced options, I haven’t talked to Mark specifically about those being passed to the backend, but they are being written to the database, so the issue there may be that they are not being picked up correctly.
Any insight from your end?

**Comment by markmfredrickson** [comment]
@rstudley: I'll take a look on the stats end.

**Comment by benthestatistician** [comment]
My experience with this arose last month when I managed to run some test jobs from the development front end. I confirmed that I had modified the advanced option and then hit Save and Continue, that the R server had received the job, and that the job spec did not include the specified within/between/across option.  So, perhaps there's an issue of R server and the web server not understanding the API in the same way. 

Alternatively, perhaps the option-not-being-passed problem is specific to the development instance, whereas Alex checked on the staging instance.  In any case I'd also flag the secondary issue noted at the top of the ticket, namely that a selected option seemed to be lost when I navigated back to Advanced Options for a second time.  That problem does seem to be internal to the front end.   

(Mark, I only got a couple test jobs to work before I started getting errors on the front and the back, leading me to think that I may have broken something else, perhaps something relating to permissions.  In other words, I wouldn't expect to be able to jump right in and reproduce the situation I was seeing.)
